### PR TITLE
feat: add `doctl` spec

### PR DESCRIPTION
**Pull request by japrozs**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Completion for the digital ocean CLI :: `doctl`

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Additional info:**

### CODE REVIEW

File: dev/doctl.ts [hunk 1]
```diff
@@ -0,0 +1,513 @@
+const DigitalOceanCTLCompletionSpec: Fig.Spec = {
+  name: "doctl",
+  description: "The official DigitalOcean command line interface (CLI)",
+  subcommands: [
+    {
+      name: "completion",
+      description:
+        "Configure your terminal's shell so that doctl commands autocomplete when you press the TAB key",
+      additionalSuggestions: ["bash", "zsh", "fish"],
```
Comment (fedeci) [inline dev/doctl.ts:9]: We should use an `arg` along with `suggestion` here:
```suggestion
      args: {
        name: "shell",
        suggestions: ["bash", "zsh", "fish"],
      },
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks, could you fix the lint errors, please? Then we should be ready to merge it!

**Comment by japrozs** [comment]
I fixed the linting changes

File: dev/doctl.ts [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+      ],
+    },
+    {
+      name: "apps",
+      description: "Display commands for working with apps",
+      subcommands: [
+        {
+          name: "",
+          description: "",
+        },
```
Comment (fedeci) [inline dev/doctl.ts:53]:
```suggestion
```

File: dev/doctl.ts [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+      name: ["-h", "--help"],
+      description: "Help for doctl",
+    },
+    {
+      name: ["-v", "--verbose"],
+      description: "Enable verbose output",
+    },
+    {
+      name: ["-o", "--output"],
+      insertValue: "json",
```
Comment (fedeci) [inline dev/doctl.ts:490]:
```suggestion
      args: {
        name: "format",
        suggestions: ["text", "json"],
      },
```

File: dev/doctl.ts [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+      description: 'Desired output format [text|json] (default "text")',
```
Comment (fedeci) [inline dev/doctl.ts:491]:
```suggestion
      description: "Desired output format",
      default: "text"
```

**Review by fedeci — CHANGES_REQUESTED** [review]

**Comment by japrozs** [comment]
more changes

**Review by fedeci — APPROVED** [review]
Thanks!

**Comment by japrozs** [comment]
No problem. Are there any other CLI's that you want specs for

**Comment by fedeci** [comment]
You can check for [issues](https://github.com/withfig/autocomplete/issues?q=is%3Aissue+is%3Aopen+label%3A%22CLI+request%22)!

**Comment by brendanfalk** [comment]
Hey @japrozs - would you be able to add support for the options for these subcommands?

e.g. `--ip-range` for `doctl vpcs`

**Comment by japrozs** [comment]
> Hey @japrozs - would you be able to add support for the options for these subcommands?
> 
> e.g. `--ip-range` for `doctl vpcs`

sure!

**Pull request merged**